Mathlib/RingTheory/Ideal/Norm/RelNorm.lean

```diff
@@ -34,44 +35,53 @@ namespace Ideal
 
 open Submodule
 
-variable (R : Type*) [CommRing R] {S : Type*} [CommRing S] [Algebra R S]
+attribute [local instance] FractionRing.liftAlgebra
 
-/-- `Ideal.spanNorm R (I : Ideal S)` is the ideal generated by mapping `Algebra.norm R` over `I`.
+variable (R S : Type*) [CommRing R] [IsDomain R] {S : Type*} [CommRing S] [IsDomain S]
+variable [IsIntegrallyClosed R] [IsIntegrallyClosed S] [Algebra R S] [Module.Finite R S]
+variable [NoZeroSMulDivisors R S] [Algebra.IsSeparable (FractionRing R) (FractionRing S)]
+
+/-- `Ideal.spanNorm R (I : Ideal S)` is the ideal generated by mapping `Algebra.intNorm R S`
+over `I`.
 
 See also `Ideal.relNorm`.
 -/
 def spanNorm (I : Ideal S) : Ideal R :=
-  Ideal.span (Algebra.norm R '' (I : Set S))
+  Ideal.span (Algebra.intNorm R S '' (I : Set S))
```

Comment:
This definition is definitely a good idea for rings of integers of number fields, because we definitely want to take the norm of an ideal there, but it also means that we lose the ability to take the norm of an ideal of S when S is finite free as an R-module but R is not an integrally closed domain. Might we one day care about this situation?

Reply:
Yes, we loose something here. I *think* it is more interesting to have the relative case, at least at the beginning. Serre does everything in the AKLB setup, and also Bourbaki seems to assume `A` integrally closed.